presto-main/src/main/java/io/prestosql/server/GracefulShutdownHandler.java

```diff
@@ -109,10 +109,10 @@ public synchronized void requestShutdown()
                 }
 
                 activeTasks = getActiveTasks();
-            }
 
-            // wait for another grace period for all task states to be observed by the coordinator
-            sleepUninterruptibly(gracePeriod.toMillis(), MILLISECONDS);
+                // wait for another grace period for all task states and shutting down state to be observed by the coordinator
+                sleepUninterruptibly(gracePeriod.toMillis(), MILLISECONDS);
```

Comment:
We need to wait at least once.... actually, I maybe we should wait first, and then shutdown once all tasks exit.

Reply:
done